projects/scion/workbench-application-platform/src/lib/manifest-capability/manifest-registry-intent-handler.service.ts

```diff
@@ -87,10 +89,16 @@ export class ManifestRegistryIntentHandler implements IntentHandler {
     const intentId = envelope.message.payload.intentId;
     const intent: Intent = this._manifestRegistry.getIntent(intentId);
 
-    const providers: Application[] = this._manifestRegistry.getCapabilities(intent.type, intent.qualifier)
+    const providers: Application[] = this._manifestRegistry.getCapabilitiesByType(intent.type)
       .filter(capability => this._manifestRegistry.isVisibleForApplication(capability, intent.metadata.symbolicAppName))
+      .filter(capability => {
+        const patchedQualifier: Qualifier = patchQualifier(intent.qualifier, capability.qualifier);
+        return matchesCapabilityQualifier(capability.qualifier, patchedQualifier);
+      })
       .map(capability => this._applicationRegistry.getApplication(capability.metadata.symbolicAppName));
-    this._messageBus.publishReply(providers, envelope.sender, envelope.replyToUid);
+
+    const distinctProviders: Application[] = Arrays.distinct(providers, (app) => app.symbolicName);
```

Comment:
Append this line as a map statement.

```
const providers: Application[] = this._manifestRegistry.getCapabilitiesByType(intent.type)
      ...
      .map(capability => this._applicationRegistry.getApplication(capability.metadata.symbolicAppName));	      
      .map(apps => Arrays.distinct(apps , app => app.symbolicName));
```

Reply:
This is not possible, as `getCapabilitiesByType` returns an array, so we'll get only a single Application in the last `map`.